app/src/main/java/com/example/atheneum/fragments/newRequest.java

```diff
@@ -99,12 +159,128 @@ public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+            public boolean onMenuItemActionExpand(MenuItem item) {
+                return true;
+            }
+
+            @Override
+            public boolean onMenuItemActionCollapse(MenuItem item) {
+                Log.d(TAG, "menu item collapse");
+                availableAdapter = new requestAdapter(newRequest.this.context, R.layout.request_list_item, defaultAvailableBook);
+                availableBookList.setAdapter(availableAdapter);
+                return true;
+            }
+        });
+
+        SearchView sv = new SearchView(((MainActivity) getActivity()).getSupportActionBar().getThemedContext());
+        MenuItemCompat.setShowAsAction(item, MenuItemCompat.SHOW_AS_ACTION_COLLAPSE_ACTION_VIEW | MenuItemCompat.SHOW_AS_ACTION_IF_ROOM);
+        MenuItemCompat.setActionView(item, sv);
+        sv.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(final String query) {
+                Log.d(TAG, "On Query Text Submit was Called");
+                db = FirebaseDatabase.getInstance();
+                ref = db.getReference("books");
+
+
+                ref.orderByChild("books").addListenerForSingleValueEvent(new ValueEventListener() {
+                    @Override
+                    public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                        searchAvailableBook.clear();
+                        for (DataSnapshot child : dataSnapshot.getChildren()) {
+                            Book book = child.getValue(Book.class);
+                            if(searchCheck(book, query)){
+                                searchAvailableBook.add(book);
+                            }
+
+                        }
+                        if (searchAvailableBook.isEmpty()) {
+                            Toast.makeText(getActivity(), "No exact matches found for search query", Toast.LENGTH_SHORT).show();
+                        }
+                        availableAdapter = new requestAdapter(newRequest.this.context, R.layout.request_list_item, searchAvailableBook);
+                        availableBookList.setAdapter(availableAdapter);
+                    }
+
+                    @Override
+                    public void onCancelled(@NonNull DatabaseError databaseError) {
+                        Log.d(TAG1, "On Cancelled of Options Menu");
+                    }
                 });
+                return false;
+            }
 
+            @Override
+            public boolean onQueryTextChange(String s) {
+                return false;
+            }
+        });
+        sv.setIconifiedByDefault(false);
+        sv.setOnSearchClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Log.d(TAG1, "On CLICKC was Called");
             }
         });
 
-        return this.view;
+        super.onCreateOptionsMenu(menu,inflater);
+    }
+
+    @Override
```

Comment:
What's the purpose of the outer onQueryTextListeners? They don't seem to do anything.